src/image.cpp

```diff
@@ -5886,7 +5888,7 @@ namespace bimg
 			depth  = bx::max<uint32_t>(1, depth);
 
 			const uint32_t mipSize = width/blockWidth * height/blockHeight * depth * blockSize;
-			const uint32_t size    = mipSize * numLayers * numSides;
+			const uint32_t size = numSides == 6 && numLayers == 1 ? mipSize * numLayers : mipSize * numSides * numLayers;
```

Comment:
What about TextureCube Array?

Reply:
The spec says

> For most textures imageSize is the number of bytes of pixel data in the current LOD level. This includes all array layers, all z slices, all faces, all rows (or rows of blocks) and all pixels (or blocks) in each row for the mipmap level. It does not include any bytes in mipPadding.
> 
> The exception is non-array cubemap textures (any texture where numberOfFaces is 6 and numberOfArrayElements is 0). For these textures imageSize is the number of bytes in each face of the texture for the current LOD level, not including bytes in cubePadding or mipPadding.

So based on that array cubemaps must use mipSize * numLayers * numSides just like everything else, as far as I can tell.

Reply:
> So based on that array cubemaps must use mipSize * numLayers * numSides just like everything else, as far as I can tell.

Now look your code (in case of cubemap, when number of layers is exactly one, you multiply `mipSize * 1`):

```cpp
const uint32_t size = numSides == 6 && numLayers == 1
    ? mipSize * numLayers
    : mipSize * numSides * numLayers
    ;
```

That code makes no sense.

Reply:
I left it in there for consistency (and was tired!) but I can change it to just mipSize. Khronos' tools doesn't really support array textures either from what I've seen so they might be difficult to validate.